doc/source/conf.py

```diff
@@ -73,13 +73,18 @@
               'sphinx.ext.ifconfig',
               'sphinx.ext.linkcode',
               'nbsphinx',
-              'sphinxcontrib.spelling'
               ]
 
 exclude_patterns = ['**.ipynb_checkpoints']
 
-spelling_word_list_filename = ['spelling_wordlist.txt', 'names_wordlist.txt']
-spelling_ignore_pypi_package_names = True
+# Tries to import sphixcontrib.spelling - if not available pass
+try:
+    import sphinxcontrib.spelling
+    extensions.append('sphinxcontrib.spelling')
+    spelling_word_list_filename = ['spelling_wordlist.txt', 'names_wordlist.txt']
+    spelling_ignore_pypi_package_names = True
+except ModuleNotFoundError:
+    pass
```

Comment:
Maybe we can print a message about no spell-check being done because it is not installed?